src/SideNav.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled, {css} from 'styled-components'
+import classnames from 'classnames'
+import {COMMON, get} from './constants'
+import theme from './theme'
+import elementType from './utils/elementType'
+import Link from './Link'
+import Box from './Box'
+import BorderBox from './BorderBox'
+
+function SideNavBase({variant, className, bordered, children, ...props}) {
+  const BoxComponent = bordered ? BorderBox : Box
+  const variantClassName = variant === 'lightweight' ? 'lightweight' : 'normal'
+  const newClassName = classnames(className, `variant-${variantClassName}`)
+
+  return (
+    <BoxComponent as="nav" className={newClassName} {...props}>
+      {children}
+    </BoxComponent>
+  )
+}
+
+const SideNav = styled(SideNavBase)`
+  background-color: ${get('colors.gray.0')};
+
+  ${props =>
+    props.bordered &&
+    css`
+      // Remove duplicate borders from nested SideNavs
+      & > & {
+        border-left: 0;
+        border-right: 0;
+        border-bottom: 0;
+      }
+    `}
+
+  ${COMMON};
+`
+
+SideNav.Link = styled(Link).attrs(props => {
+  const isReactRouter = typeof props.to === 'string'
+  if (isReactRouter || props.selected) {
+    // according to their docs, NavLink supports aria-current:
+    // https://reacttraining.com/react-router/web/api/NavLink/aria-current-string
+    return {'aria-current': 'page'}
+  } else {
+    return {}
+  }
+})`
+  position: relative;
+  display: block;
+  ${props =>
+    props.variant === 'full' &&
+    css`
+      display: flex;
+      align-items: center;
+      justify-content: space-between;
+    `}
+  width: 100%;
+  text-align: left;
+  // TOOD [MKT]: this wasn't in the Primer CSS version but
+  // is required to make the component match visually
+  font-size: ${get('fontSizes.1')};
```

Comment:
@emplums Did we decide the best way to handle this?

Reply:
For now let's keep setting the font in each component 🤔We should spend some time thinking about whether or not components should set their own font sizes or inherit them from a font size declared on the body, given we don't have much control over whether or not someone uses the `BaseStyles` component. Could you open a discussion issue for that?